web3.js/test/transaction.test.ts

```diff
@@ -365,14 +365,14 @@ describe('Transaction', () => {
     }).add(transfer);
     expectedTransaction.sign(sender);
 
-    const wireTransaction = Buffer.from(
+    const serializedTransaction = Buffer.from(
       'AVuErQHaXv0SG0/PchunfxHKt8wMRfMZzqV0tkC5qO6owYxWU2v871AoWywGoFQr4z+q/7mE8lIufNl/kxj+nQ0BAAEDE5j2LG0aRXxRumpLXz29L2n8qTIWIY3ImX5Ba9F9k8r9Q5/Mtmcn8onFxt47xKj+XdXXd3C8j/FcPu7csUrz/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAxJrndgN4IFTxep3s6kO0ROug7bEsbx0xxuDkqEvwUusBAgIAAQwCAAAAMQAAAAAAAAA=',
       'base64',
     );
-    const tx = Transaction.from(wireTransaction);
+    const deserializedTransaction = Transaction.from(serializedTransaction);
 
-    expect(tx).to.eql(expectedTransaction);
-    expect(wireTransaction).to.eql(expectedTransaction.serialize());
+    expect(expectedTransaction.serialize()).to.eql(serializedTransaction);
+    expect(deserializedTransaction.serialize()).to.eql(serializedTransaction);
   });
 
   it('populate transaction', () => {
```

Comment:
It's probably too much work, but adding a few of these might prevent someone from breaking this code in the future.

```
// Do some illegal modification...
expect(transaction.compileMessage).to.throw();
```

Reply:
I agree that we should have test coverage that a post-populate modification actually throws an error